src/libpgmoneta/wal.c

```diff
@@ -171,6 +171,10 @@ pgmoneta_wal(int srv, char** argv)
       memset(xlogpos, 0, xlogpos_size);
       memcpy(xlogpos, pgmoneta_query_response_get_data(identify_system_response, 2), xlogpos_size);
 
+      // Assign the current xlogpos to LSN
+      memset(config->servers[srv].pgmoneta_current_wal_lsn, 0, MISC_LENGTH);
+      snprintf(config->servers[srv].pgmoneta_current_wal_lsn, MISC_LENGTH, "%s", xlogpos);
```

Comment:
I would prefer you move this line downwards, after we have validated it by wal_convert_xlogpos.

Reply:
I had this block of code after the wal_convert_xlogpos if-block originally. wal_convert_xlogpos use strtok on xlogpos by '/' character. From what I understand about strtok, after the wal_convert_xlogpos exits, xlogpos will become '0' instead of '0/19FCACO' for example. 0 was also displayed as LSN on Prometheus metrics when I ran the code. So I moved that block up.

Reply:
Nice catch! Yeah you are right. Then just leave it there, I'll submit a patch to improve the xlogpos parsing and move it down myself later.

Reply:
Got it.